### PR TITLE
Fixing desired capabilities and find element deprecation

### LINKED ISSUE #1870: "DeprecationWarning: desired_capabilities  has been deprecated, please pass in a Service object" when running tests
I'm getting a `DeprecationWarning` when running tests for a dash app. A complete minimal working example would be

## minimal test app

```python
# usage_tmp.py

import dash
from dash import html

app = dash.Dash(__name__)
app.layout = html.Div([html.H1("test")])

if __name__ == "__main__":
    app.run_server(debug=True)

```

## minimal test script

```python
# tests\test_usage_tmp.py

from dash.testing.application_runners import import_app

# Run with pytest -k testing123
def test_testing123(dash_duo):
    app = import_app("usage_tmp")

```

## Example output

```
(myproj) PS C:\Users\USER\repos\myproj> python -m pytest -k testing123
Test session starts (platform: win32, Python 3.8.6, pytest 6.2.5, pytest-sugar 0.9.4)
cachedir: .pytest_cache
rootdir: C:\Users\USER\repos\myproj, configfile: pytest.ini, testpaths: tests/
plugins: dash-2.0.0, flaky-3.7.0, mock-3.6.1, sugar-0.9.4
collecting ... 
 tests\test_usage_tmp.py::test_testing123 ✓                                                                            100% ██████████
========================================================= warnings summary ========================================================== 
tests/test_usage_tmp.py::test_testing123
  C:\Python\venvs\myproj\lib\site-packages\selenium\webdriver\chrome\webdriver.py:70: DeprecationWarning: desired_capabilities 
has been deprecated, please pass in a Service object
    super(WebDriver, self).__init__(DesiredCapabilities.CHROME['browserName'], "goog",

-- Docs: https://docs.pytest.org/en/stable/warnings.html
```

## Versions
- dash  2.0.0
- selenium 4.1.0

### PR DESCRIPTION
**Pull request by MrTeale**
This PR closes #1870 

When running the tests, `DeprecationWarning: desired_capabilities has been deprecated, please pass in a Service object` is shown.

There was also another similar message that has been dealt with in this PR: `DeprecationWarning: find_element_by_* commands are deprecated. Please use find_element() instead`. This has been solved by `find_element` and `find_elements` now taking an attribute keyword, as a string, for the type of selector that is being used. default is set to `CSS_SELECTOR` as that's what historically `find_element` was solely for.

## Contributor Checklist

- [x] I have broken down my PR scope into the following TODO tasks
   -  [x] Fix `desired_capabilities` deprecation warning
   -  [x] Fix `find_element_by_*` deprecation warning
- [x] I have run the tests locally and they passed. (refer to testing section in [contributing](https://github.com/plotly/dash/blob/master/CONTRIBUTING.md))
- [x] I have added entry in the `CHANGELOG.md`

### CODE REVIEW

**Comment by MrTeale** [comment]
This is currently causing some tests to be flaky. Looking into why a bit more so until then, PLEASE DO NOT MERGE.

 If you have any questions or feedback, let me know.

File: dash/testing/browser.py
Comment (alexcjohnson) [inline dash/testing/browser.py:233]: Seems reasonable - let's just describe `attribute` in the docstring (here and `find_elements`) along with the list of acceptable values, which (from `dir(By)`) appears to be:
```
'CLASS_NAME', 'CSS_SELECTOR', 'ID', 'LINK_TEXT', 'NAME', 'PARTIAL_LINK_TEXT', 'TAG_NAME', 'XPATH'
```

File: tests/integration/devtools/test_devtools_ui.py
Comment (alexcjohnson) [inline tests/integration/devtools/test_devtools_ui.py:259]:
```suggestion
    dash_br.find_element("#alternative_id")
```
I don't mind giving folks the option to use `ID` and others, but within this repo, when a CSS selector is easy I'd rather we standardize.

**Review by alexcjohnson — APPROVED** [review]
💃 Looks great, thanks for the contribution @MrTeale 

The `table-server-test` failure here is nothing to do with your changes, it's an intermittent failure we've seen recently, so far we haven't figured out the origin but it shouldn't block your PR.

File: tests/integration/devtools/test_devtools_ui.py
Comment (MrTeale) [reply tests/integration/devtools/test_devtools_ui.py]: I totally agree. Using the ID css selector totally slipped my mind. Great pickup @alexcjohnson. I've made that change.

File: dash/testing/browser.py
Comment (MrTeale) [reply dash/testing/browser.py]: Added a list of acceptable `attribute` values to the docstring and noted that they align with the Selenium API's `By` class.

**Comment by MrTeale** [comment]
Thanks @alexcjohnson! This should be ready to go now

**Pull request merged**
